test-specifications/token-service/tokenMintTransaction.md

```diff
@@ -0,0 +1,190 @@
+This test specification for TokenMintTransaction is to be one of many for testing the functionality of the Hedera SDKs. The SDK under test will use the language specific JSON-RPC server return responses back to the test driver.
+
+## Design:
+Each test within the test specification is linked to one of the properties within TokenMintTransaction. Each property is tested with a mix of boundaries. The inputs for each test are a range of valid, minimum, maximum, negative and invalid values for the method. The expected response of a passed test can be a correct error response code or seen as the result of node queries. A successful transaction (the transaction reached consensus and was applied to state) can be determined by getting a `TransactionReceipt` or `TransactionRecord`, or can be determined by using queries such as `TokenInfoQuery` or `AccountBalanceQuery` and investigating for the required changes (creations, updates, etc.). The mirror node can also be used to determine if a transaction was successful via its rest API. Error codes are obtained from the response code proto files.
+
+**Transaction properties:**
+
+https://docs.hedera.com/hedera/sdks-and-apis/sdks/token-service/mint-a-token
+
+**TokenMint protobufs:**
+
+https://github.com/hashgraph/hedera-protobufs/blob/main/services/token_mint.proto
+
+**Response codes:**
+
+https://github.com/hashgraph/hedera-protobufs/blob/main/services/response_code.proto
+
+**Mirror Node APIs:**
+
+https://docs.hedera.com/hedera/sdks-and-apis/rest-api
+
+## JSON-RPC API Endpoint Documentation
+
+### Method Name
+
+`mintToken`
+
+### Input Parameters
+
+| Parameter Name          | Type                                             | Required/Optional | Description/Notes                                                 |
+|-------------------------|--------------------------------------------------|-------------------|-------------------------------------------------------------------|
+| tokenId                 | string                                           | optional          | The ID of the token to mint.                                      |
+| amount                  | string                                           | optional          | The amount of fungible tokens to mint.                            |
+| metadata                | list<string>                                     | optional          | The metadata for the non-fungible tokens to mint, as hex strings. |
+| commonTransactionParams | [json object](../commonTransactionParameters.md) | optional          |                                                                   |
+
+### Output Parameters
+
+| Parameter Name | Type   | Description/Notes                                                                 |
+|----------------|--------|-----------------------------------------------------------------------------------|
+| status         | string | The status of the submitted `TokenMintTransaction` (from a `TransactionReceipt`). |
```

Comment:
Might be beneficial to also return the `nftIds` for the minted nfts.

Reply:
Good call! Added in `newTotalSupply` as well